File: code/java/DataStructure/src/lib/AVLTree/AVLNode.java
package AVLTree;

public class AVLNode {
    private int val;
    private AVLNode right;
    private AVLNode left;
    private int height;

    public AVLNode(int val) {
        this.val = val;
        this.right = null;
        this.left = null;
        this.height = 1;
    }


    public int getVal() {
        return val;
    }

    public void setVal(int val) {
        this.val = val;
    }

    public AVLNode getRight() {
        return right;
    }

    public void setRight(AVLNode right) {
        this.right = right;
    }

    public AVLNode getLeft() {
        return left;
    }

    public void setLeft(AVLNode left) {
        this.left = left;
    }

    public int getHeight() {
        return height;
    }

    public void setHeight(int height) {
        this.height = height;
    }
}

File: code/go/avl/avl_node.go
package avl

type avlNode struct {
	key    int32
	height int32
	left   *avlNode
	right  *avlNode
}

func newAvlNode(key int32) *avlNode {
	return &avlNode{key: key, height: 0, left: nil, right: nil}
}
